src/App.tsx

```diff
@@ -16,35 +22,77 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibileGoods = [...goodsFromServer];
+
+  if (sortField) {
+    visibileGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibileGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button ${sortField === SortType.ALPHABETICALLY ? 'is-info' : 'is-light'}`}
+          onClick={() => setSortField(SortType.ALPHABETICALLY)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button ${sortField === SortType.LENGTH ? 'is-success' : 'is-light'}`}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button ${isReversed ? 'is-warning' : 'is-light'}`}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== SortType.DEFAULT || isReversed) && (
+          <button
+            type="button"
+            className={`button is-danger is-light`}
+            onClick={() => {
+              setSortField(SortType.DEFAULT);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibileGoods.map(good => (
```

Comment:
Ensure that the variable `visibileGoods` is consistently renamed to `visibleGoods` throughout the file to maintain consistency.